tests/cbtest/gtest/test_export.cc

```diff
@@ -0,0 +1,267 @@
+#include <iostream>
+#include <libuzfs.h>
+#include <libzfs.h>
+#include <uzfs_mgmt.h>
+#include <zrepl_mgmt.h>
+#include <uzfs_io.h>
+#include <system_error>
+
+#include <gtest/gtest.h>
+
+#define DISK "/tmp/zrepl_test_disk.img"
+#define POOL_NAME "test_export"
+
+int
+create_file(void)
+{
+       int fd;
+
+       if ((fd = open(DISK, O_RDWR | O_CREAT | O_TRUNC, 0666)) < 0)
+               throw std::runtime_error("cant create disk image");
+
+       if (ftruncate(fd, 100 * 1024 * 1024) != 0)
+               throw std::runtime_error("cant truncate disk image");
+       close(fd);
+       return (0);
+}
+
+/*
+ * if you find the need to copy paste the following
+ * two functions consider to put them in a generic
+ * header file, because I already copied them =)
+ *
+ */
+
+std::string getCmdPath(std::string zfsCmd) {
+	std::string cmdPath;
+	const char *srcPath = std::getenv("SRC_PATH");
+
+	if (srcPath == NULL) {
+		cmdPath += ".";
+	} else {
+		cmdPath = srcPath;
+	}
+	cmdPath += "/cmd/" + zfsCmd + "/" + zfsCmd;
+
+	return cmdPath;
+}
+
+void execCmd(std::string zfsCmd, std::string args) {
+	int rc;
+	std::string cmdLine;
+
+	cmdLine = getCmdPath(zfsCmd) + " " + args;
+	rc = system(cmdLine.c_str());
+	if (rc != 0) {
+		throw std::runtime_error(
+		    std::string("Command failed: ") + cmdLine);
+	}
+}
+
+class CreatePool : public testing::Test {
+public:
+
+	spa_t *spa;
+	CreatePool() : zi(NULL), zv(NULL), spa(NULL) {}
+
+	virtual void SetUp() override {
+		uzfs_init();
+		libuzfs_ioctl_init();
+	}
+
+
+	void createPool() {
+		ASSERT_TRUE(uzfs_create_pool(pool_name, DISK, &spa ) == 0);
```

Comment:
this PR being to fix CLI, its better to create pool/datasets through CLI, by having 'zrepl' running

Reply:
These tests pool related feature without zrepl -- and we should not depend on zrepl as it "just an application" for running any test cases against core libzpool features.

Reply:
This PR tests CLI and API functions BTW.

Reply:
hmm.. I see CLI for zpool destroy/export,  and zfs create. But, pool is being created and imported using APIs which won't be the case usually.
And, code is duplicated for import, which actually tests unit test code rather than production code.